### PR TITLE
Initial Commit for UDA job-Postgres

### PR DESCRIPTION
**Pull request by naveena-maplelabs**

### CODE REVIEW

File: playbooks/create_uda_job.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      cohesity_uda_job:
+        state: present
+        job_name: protect_uda
+        storage_domain: "DefaultStorageDomain"
+        policy: "Silver"
+        endpoint: "x.x.x.x"
+        cancel_active: true
+        objects:
+          - object
+        cluster: "{{ cohesity_server }}"
```
Comment (pyashish) [inline playbooks/create_uda_job.yml:18]: Make this part of inventory instead of the playbook

File: plugins/modules/cohesity_uda_job.py [hunk 2]
```diff
@@ -0,0 +1,702 @@
...
+        cmd = "timedatectl status"
+        proc = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, stdin=subprocess.PIPE, shell=True
+        )
+        out, err = proc.communicate()
+        for line in out.split("\n"):
+            if "Time zone" in line:
+                default_timezone = line.split()[2]
+    except Exception:
+        pass
```
Comment (pyashish) [inline plugins/modules/cohesity_uda_job.py:222]: Do not leave exception handling this way, use ansible modules to catch exceptions or to throw valid ones.

File: plugins/modules/cohesity_uda_job.py [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+        if status == "kAccepted":
+            return True, status, last_run
+        elif status in ["kCanceled", "kSuccess"]:
+            return False, status, last_run
+        return False, status, ""
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def check__protection_job__exists(module):
```
Comment (pyashish) [inline plugins/modules/cohesity_uda_job.py:319]: can we have consistency between single underscores and double ?

**Review by pyashish — CHANGES_REQUESTED** [review]
Fix lint errors too.

File: plugins/modules/cohesity_uda_job.py [hunk 4]
```diff
@@ -0,0 +1,702 @@
+#!/usr/bin/python
+# Copyright (c) 2022 Cohesity Inc
+
+
+from __future__ import absolute_import, division, print_function
```
[the remark this answers is not in the document]
Comment (naveena-maplelabs) [reply plugins/modules/cohesity_uda_job.py:5]: There are mandatory lines added in all the modules.

File: playbooks/create_uda_job.yml (hunk 1 above)
Comment (naveena-maplelabs) [reply playbooks/create_uda_job.yml:18]: We have the declared cohesity_server variable in the inventory file, we will refer the variable here.

**Comment by naveena-maplelabs** [comment]
@pyashish Addressed the comments and  updated the module. please review.

File: plugins/modules/cohesity_uda_job.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pyashish) [reply plugins/modules/cohesity_uda_job.py]: Whats the purpose if we aren't using ? remove it from now on, other ones, we can clean it up later

**Comment by naveena-maplelabs** [comment]
In all the modules `from __future__ import (absolute_import, division, print_function)` is required.


naveena.maplelabs@u18-101648-01:~/ansible_collections/cohesity/dataprotect$ **ansible-test sanity plugins/modules/cohesity_job.py** 
Running sanity test 'action-plugin-docs' with Python 3.6
Running sanity test 'empty-init' with Python 3.6
Running sanity test 'future-import-boilerplate' with Python 3.6
```
ERROR: Found 1 future-import-boilerplate issue(s) which need to be resolved:
ERROR: plugins/modules/cohesity_job.py:0:0: missing: from __future__ import (absolute_import, division, print_function)

```

**Comment by pyashish** [comment]
Fix lint errors and merge please.

**Pull request merged**